vms/platformvm/block/builder/builder.go

```diff
@@ -244,14 +246,39 @@ func (b *builder) PackAllBlockTxs() ([]*txs.Tx, error) {
 		return nil, fmt.Errorf("%w: %s", errMissingPreferredState, preferredID)
 	}
 
-	return packBlockTxs(
+	timestamp, _, err := state.NextBlockTime(preferredState, b.txExecutorBackend.Clk)
+	if err != nil {
+		return nil, fmt.Errorf("could not calculate next staker change time: %w", err)
+	}
+
+	if !b.txExecutorBackend.Config.UpgradeConfig.IsEtnaActivated(timestamp) {
+		return packDurangoBlockTxs(
+			preferredID,
+			preferredState,
+			b.Mempool,
+			b.txExecutorBackend,
+			b.blkManager,
+			timestamp,
+			math.MaxInt,
+		)
+	}
+
+	stateDiff, err := state.NewDiffOn(preferredState)
+	if err != nil {
+		return nil, err
+	}
+
+	if _, err := txexecutor.AdvanceTimeTo(b.txExecutorBackend, stateDiff, timestamp); err != nil {
```

Comment:
I think this was referencing an older version of the PR. It was done this way because the `Etna` block packing needs to honor `Capacity` which is based on the block state... Whereas the `Durando` logic just enforced a flat maximum size.

This was changed to just specify a `minCapacity` value which is set differently during block building or mempool pruning.